docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -635,310 +960,179 @@ public class CustomSecurityIdentityAugmentor implements SecurityIdentityAugmento
-quarkus.oidc.credentials.jwt.key-password=mykeypassword
+@ApplicationScoped
+public class SecurityEventListener {
 
-# Private key alias inside the keystore
-quarkus.oidc.credentials.jwt.key-id=mykeyAlias
+    public void event(@Observes SecurityEvent event) {
+        String tenantId = event.getSecurityIdentity().getAttribute("tenant-id");
+        RoutingContext vertxContext = event.getSecurityIdentity().getAttribute(RoutingContext.class.getName());
+        vertxContext.put("listener-message", String.format("event:%s,tenantId:%s", event.getEventType().name(), tenantId));
+    }
+}
 ----
 
-Using `client_secret_jwt` or `private_key_jwt` authentication methods ensures that no client secret goes over the wire.
+=== Propagating tokens to downstream services
 
-==== Additional JWT authentication options
+For information about Authorization Code Flow access token propagation to the downstream services, see the xref:security-openid-connect-client-reference.adoc#token-propagation[Token Propagation] section.
 
-If `client_secret_jwt`, `private_key_jwt` authentication methods are used or an Apple `post_jwt` method is used, then the JWT signature algorithm, key identifier, audience, subject and issuer can be customized, for example:
 
-[source,properties]
-----
-# private_key_jwt client authentication
+== Integration considerations
 
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
-quarkus.oidc.credentials.jwt.key-file=privateKey.pem
+Your OIDC backed application integrates in an environment such as calling from a single page application, using well known OIDC providers etc. 
+This section discusses these considerations.
 
-# This is a token key identifier 'kid' header - set it if your OpenID Connect provider requires it.
-# Note if the key is represented in a JSON Web Key (JWK) format with a `kid` property then
-# using 'quarkus.oidc.credentials.jwt.token-key-id' is not necessary.
-quarkus.oidc.credentials.jwt.token-key-id=mykey
+=== Single-page applications
 
-# Use RS512 signature algorithm instead of the default RS256
-quarkus.oidc.credentials.jwt.signature-algorithm=RS512
+Check if implementing SPAs the way it is suggested in the xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[Single-page Applications for Service Applications] section can meet your requirements.
 
-# The token endpoint URL is the default audience value, use the base address URL instead:
-quarkus.oidc.credentials.jwt.audience=${quarkus.oidc-client.auth-server-url}
+If you prefer to use SPA and JavaScript API such as `Fetch` or `XMLHttpRequest`(XHR) with Quarkus web applications, be aware that OpenID Connect Providers may not support CORS for Authorization endpoints where the users are authenticated after a redirect from Quarkus. This will lead to authentication failures if the Quarkus application and the OpenID Connect Provider are hosted on the different HTTP domains/ports.
```

Comment:
```suggestion
If you prefer to use SPAs and JavaScript APIs such as `Fetch` or `XMLHttpRequest`(XHR) with Quarkus web applications, be aware that OpenID Connect providers might not support cross-origin resource sharing (CORS) for authorization endpoints where the users are authenticated after a redirect from Quarkus. 
This will lead to authentication failures if the Quarkus application and the OpenID Connect provider are hosted on different HTTP domains, ports, or both.
```